### ISSUE TITLE
Colorspace parsing error

### ISSUE THREAD

**Issue by pizzaow** [mozilla/pdf.js#6081]
Document fails to load:

Error: unimplemented color space object "undefined" 

```
Error: unimplemented color space object "undefined"
/pdf.js/build/pdf.worker.js:244     at error (https://mozilla.github.io/pdf.js/build/pdf.worker.js:244:17)
    at Function.ColorSpace_parseToIR (https://mozilla.github.io/pdf.js/build/pdf.worker.js:6873:11)
    at Function.ColorSpace_parse (https://mozilla.github.io/pdf.js/build/pdf.worker.js:6718:25)
    at next (https://mozilla.github.io/pdf.js/build/pdf.worker.js:11342:28)
    at Object.PartialEvaluator_getOperatorList [as getOperatorList] (https://mozilla.github.io/pdf.js/build/pdf.worker.js:11221:14)
    at https://mozilla.github.io/pdf.js/build/pdf.worker.js:2768:33
```

PDFJS.version = '1.1.181';
PDFJS.build = '82c5cf6';

I believe the relevant parts of the problem are:
```
5 0 obj
<< 
   [...]
   /Resources << /ColorSpace << /CS0 6 0 R >> /Pattern << /P0 8 0 R >> >>
   [...]
>>
endobj

6 0 obj
   [ 7 0 R ]
endobj

7 0 obj
   /Pattern
endobj
```
Looking at the code, I suspect the problem is in ColorSpace.parseToIR:
```
cs = xref.fetchIfRef(cs);
At the point cs looks to be:
cs: Array[1] {
  0: Ref {
    gen:0
    num: 11
  }
  length :1
}
```
'Pattern' is not getting picked up properly (either as a .name or cs[0].name

Example PDF located here: 
http://csaim.com/bad_example.pdf

As a temporary workaround, I changed the switch 'default' to behave like 'Pattern', which gets this particular PDF to display.

**Comment by pizzaow** [comment]
Example PDF located here: 
http://csaim.com/bad_example.pdf

**Comment by Snuffleupagus** [comment]
@pizzaow Are you the creator of the PDF file, and if so can we include it in our test suite?

**Comment by pizzaow** [comment]
Yes, I am the creator. And yes, you are welcome to include this file in your test suite.

**Comment by Snuffleupagus** [comment]
> Yes, I am the creator. And yes, you are welcome to include this file in your test suite.

Thank you; I've submitted a patch that fixes this issue, see #6082.

**Comment by pizzaow** [comment]
Wow, very quick response! I tested your change on my development computer with some additional examples and it does resolve the problem!
